cmd/upload.go

```diff
@@ -80,9 +80,17 @@ func init() {
 					progressBar.Add(1)
 				}
 
-				if branch == "" && git.Branch == "" {
+				// either both commit and branch are specified or both are unspecified
+				// fail otherwise
+				commitProvided := bool(git != nil && git.Commit != "")
+				branchProvided := bool(git != nil && git.Branch != "" || branch != "")
```

Comment:
I think operator precedence may bite you here, with the && higher than the ||

Reply:
I've rewritten it as `(git != nil && git.Branch != "") || branch != ""` for clarity. Functionally, it should be the same.